### PR TITLE
#184637120 postgresDB and sequelize ORM configuration

### PR DESCRIPTION
**Pull request by katros1**
#### What does this PR do?

- Configuration of postgresDB and sequelize ORM

#### Description of Task to be completed?

- Configure The App to use PostgreSQL as DB Engine and Sequelize ORM

#### How should this be manually tested?

- clone  the repository 
- checkout to develop branch
- pull from the develop branch
- checkout to the new branch you want to work in 
- to install the dependencies used run in your terminal ` npm install `
- to run app run in your terminal ` npm run dev `

#### To test the Database

- first create .env file
- check .env.example how to add your database url
- In you terminal  navigate to database folder by running `cd src/database`
- In database folder you can run  `sequelize db:migrate to create table`

NOTE: in database folder is where to create tables

#### What are the relevant pivotal tracker strory?

#184637120

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,11 @@
 {
   "name": "destructors-ec-bn",
   "version": "1.0.0",
-  "type": "module",
   "description": "Destructors ecommerce team project",
   "main": "index.js",
   "scripts": {
-    "start": "node src/index.js",
-    "dev": "nodemon src/index.js"
+
```
Comment (leandreAlly) [inline package.json:7]: You removed start script for production

Comment (leandreAlly) [inline package.json]: You had removed start script for production

File: src/database/config/config.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require('dotenv').config();
+
+module.exports = {
+  development: {
+    url: 'postgres://postgres:babu02@127.0.0.1:5432/ecdev_db',
```
Comment (leandreAlly) [inline src/database/config/config.js]: remove your database url and use environment variable to allow everyone
url: 'proccess.env.DEV_DATABASE_URL'

File: package.json (hunk 1 above)
Comment (leandreAlly) [reply package.json]: also you can save time by creating script command for migrating database 
"migrate": "sequelize db:migrate"

Comment (katros1) [reply package.json]: thank you @leandreAlly,

File: src/database/config/config.js (hunk 2 above)
Comment (katros1) [reply src/database/config/config.js]: ooh!!! I forget to change, thanks @leandreAlly

**Review by leandreAlly — APPROVED** [review]

**Review by leandreAlly — CHANGES_REQUESTED** [review]
create a functions that connect the database, cause we need to know if database it connected or not?.
make sure you allow server to run when database is connected.

**Review by mjeanbosco19 — APPROVED** [review]
Great Job @katros1

**Review by mjeanbosco19 — CHANGES_REQUESTED** [review]
I recommend you To "squash" in Git means to combine your multiple commits into one commit or at least not above two commits.

File: src/database/migrations/20230309071643-modify_users_add_new_fields.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  },
+
+
+  async down (queryInterface, Sequelize) {
+    /**
+     * Add reverting commands here.
+     *
+     * Example:
+     * await queryInterface.dropTable('users');
+     */
```
Comment (Abdulkeza) [inline src/database/migrations/20230309071643-modify_users_add_new_fields.js:33]: why this??

**Review by buka1calvin — APPROVED** [review]
on my side i've tested your work i think you're ready to go thank you

**Comment by katros1** [comment]
> create a functions that connect the database, cause we need to know if database it connected or not?. make sure you allow server to run when database is connected.

thanks to @leandreAlly for this suggestion, we worked on it and now whenever we run our server we can see if the database connection is successful or failed.

**Comment by katros1** [comment]
> I recommend you To "squash" in Git means to combine your multiple commits into one commit or at least not above two commits.

thanks @mjeanbosco19 for the reminder,  now we have on commit

**Comment by katros1** [comment]
> on my side i've tested your work i think you're ready to go thank you

thanks @buka1calvin

**Review by Abdulkeza — APPROVED** [review]

File: src/database/migrations/20230309071643-modify_users_add_new_fields.js (hunk 3 above)
Comment (katros1) [reply src/database/migrations/20230309071643-modify_users_add_new_fields.js:33]: Ooh!! These are default codes which are in the files which are created when `sequelize-cli migration:create --name modify_users_add_new_fields`  command is run forinstance when you want to modify the table or to add a field. So they guide how to edit those files

**Review by ba-lambert — APPROVED** [review]

**Pull request merged**